database/scrapers/contributors.ts

```diff
@@ -142,26 +140,15 @@ export default class ContributorsScraper extends BaseScraperModule {
       await Promise.all(
         Array.from(imageIds).map((id) =>
           this.semaphore.runTask(async () => {
-            const details = (await this.fetchJSON(
-              `https://admin.topwr.solvro.pl/files/${id}?fields=filename_disk`,
-              `details for file ${id}`,
-            )) as { data: { filename_disk: string } };
-            const extension = details.data.filename_disk
-              .split(".")
-              .pop()
-              ?.toLowerCase();
-            const fileResponse = await this.fetchAndCheckStatus(
-              `https://admin.topwr.solvro.pl/assets/${id}`,
-              `file ${id}`,
-            );
-            if (fileResponse.body === null) {
-              throw new Error(`Response for file ${id} has no body, wtf`);
-            }
-            const file = await FilesService.uploadStream(
-              Readable.fromWeb(fileResponse.body),
-              extension,
-            ).addErrorContext(() => `Failed to upload file ${id} to storage`);
-            return [id, file.id] as [string, string];
+            return [
+              id,
+              await this.directusUploadFieldAndGetKey(id).then((fileId) => {
+                if (fileId === null) {
+                  throw new Error(`Failed to upload file ${id}`);
+                }
+                return fileId;
+              }),
```

Comment:
what did you do to my beautiful error handling
this is why you don't return fake values as failures - just throw an error, you could've just used `.addErrorContext` and you would've preserved full cause information